File: Day-13-of-30-Days/Solution.py
def calculate_vowels(input_str):
    vowel_counts = {}
    vowels = "aeiou"

    for vowel in vowels:
        vowel_counts[vowel] = 0

    input_str = input_str.lower()
    for char in input_str:
        if char in vowels:
            vowel_counts[char] += 1

    return vowel_counts

print(calculate_vowels("Hello World"))
print(calculate_vowels("Try"))

File: Day-13-of-30-Days/Solution.cpp
// Day 13 of 30 Days 👾
#include <iostream>
#include <unordered_map>
#include <string>
    using namespace std;

unordered_map<char, int> calculateVowels(const string& input) {
    unordered_map<char, int> vowelCounts;
    string vowels = "aeiou";

    for (char vowel : vowels) {
        vowelCounts[vowel] = 0;
    }

    string lowercaseInput = input;
    // Convert input string to lowercase
    for (char& c : lowercaseInput) {
        c = tolower(c);
    }

    for (char c : lowercaseInput) {
        if (vowels.find(c) != string::npos) {
            vowelCounts[c]++;
        }
    }

    return vowelCounts;
}

int main() {
    auto result1 = calculateVowels("Hello World");
    auto result2 = calculateVowels("Try");

    // Examples
    for (const auto& pair : result1) {
        cout << pair.first << ": " << pair.second << endl;
    }

    for (const auto& pair : result2) {
        cout << pair.first << ": " << pair.second << endl;
    }

    return 0;
}